for_curators/sample_local_dataset_upload_dev.py.ipynb

```diff
@@ -0,0 +1,305 @@
+{
+ "cells": [
+  {
+   "cell_type": "markdown",
+   "id": "b68160f6",
+   "metadata": {},
+   "source": [
+    "# Upload a local datafile to add or replace a Dataset in a Collection"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "1ff1ec6e",
+   "metadata": {},
+   "source": [
+    "#### This script performs the upload of a local datafile to a given Collection (as identified by its Collection uuid), where the datafile becomes a Dataset accessible via the Data Portal UI. In order to use this script, you must have a Curation API key, you must know the id of the Collection to which you wish to upload the datafile, and you must decide upon a string tag (the `curator_tag`) to use to uniquely identify the resultant Dataset within this Collection going forward. Uploads to a tag that has not been used yet in the given Collection will result in a new Dataset being created. Uploads to a tag for which there already exists a Dataset in the given Collection will result in the existing Dataset being replaced by the new Dataset created from the datafile that you are uploading. You can only add/replace Datasets in private Collections or revision Collections."
```

Comment:
Nit: The `####` feels a bit heavy. 😄 Why bold the text of the entire paragraph?

Reply:
For completeness, we should probably include instructions on how to obtain the curation API key.

Reply:
Maybe add a `TODO` to add instructions for checking the ingestion status of the dataset once the status endpoint is available via the curation API.